### DISCUSSION TITLE
Make Load Game open the dialog on the adventure map instead of returning to the Main Menu

### DISCUSSION THREAD

**Discussion by zenseii** [ihhub/fheroes2#8472] [Ideas]
I'm wondering what others think about changing the Load Game dialog so that it opens directly in the Adventure Map view instead of sending you to the Main Menu. I know this is how the original game works, and the only "benefit" to this is that you can (ex.) more quickly load a Campaign game when currently playing a Standard game, but I don't see why that would be something you do often.

The behavior I'm suggesting is to just open the respective list of save games for the game type you're currently playing.

<img width="371" alt="image" src="https://github.com/ihhub/fheroes2/assets/12501091/8a122bca-0588-4dc3-82b0-0ad10f9ed86e">

It would be like this:
<img width="687" alt="image" src="https://github.com/ihhub/fheroes2/assets/12501091/041030f8-2d9d-4893-aa10-9c4c65497055">

Any opinions on this, @Branikolog, @Districh-ru, @ihhub, @oleg-derevenetz ?

**Comment by ihhub** [comment]
Hi @zenseii , the only problem I see that there would be no choice to select a save for other types of games such as campaigns while playing single-player. However, I would be happy to have this dialog over every time going to New Menu.

**Reply by zenseii** [reply]
Yes I agree. I think the amount of times that players are playing a standard single player game and then all of a sudden want to load into a campaign game, are negligible. For the rare times they want to do that, they can return to the main menu with "New Game".

**Reply by Districh-ru** [reply]
I agree that it will be more convenient no to go every time to the main menu and select the game type - mostly it is the current game type you are playing.
We can make a new Load dialog and add new buttons to switch game type from this dialog (in future).

**Comment by oleg-derevenetz** [comment]
As for me, I am quite satisfied with the original logic :) At least it is designed to be able to choose the game type to load, and this is what users of the original game are used to.

**Reply by AlexSnowLeo** [reply]
It would be cool to implement this as an optional option in the settings, or even as a mod.

**Reply by zenseii** [reply]
@oleg-derevenetz. Thanks for your input. It's a very valid point that the current behavior is what users know, and we don't get any complaints about it so why "fix" something which isn't broken basically. Anyways I just wanted to throw this idea and hear what people think about it. At least now if someone comes and proposes it we can direct them here.

**Comment by Branikolog** [comment]
Hi, @zenseii 
I'm OK with current implementation.
I even consider such interaction as a penalty for players, who's about to "cheat" by save/loading. Making it easier to reload the game can kill all the challenge and at the same time make the game results not so exciting, as player made thousands of reloads to achieve the perfect (not possible in common case, as we're all humans and make mistakes sometimes 😄 ). 
Uncontrolled (~easy) reloads could even spoil game experience and make player bored.
A small "uncomfortable" action required for reloading encourages player thinking like: "Ah, ok... I made some mistakes but the result of the battle (turn) is acceptable. I'm too lazy to exit to menu and reaload the game once again... Lets play further!"
Same happens in console games, when a player can reload the game state only from the certain checkpoints. That forces him to continue playing as it is, otherwize he should replay the game from checkpoint just to reach the same game state at the moment.

**Reply by zenseii** [reply]
I see and I agree with your point about not incentivizing players to save-reload spamming. I appreciate you taking your time to give the feedback and I obviously opened this discussion instead of filing a issue because I wanted to hear other people's opinion on it.

I'm very used to the way it currently works and can't really complain. I'm not sure if there is a reason to compare with later games. I believe Heroes 4 does what I described, not sure about 3, and I personally can't say it makes me more inclined to continue playing, but we need to think about what the most players' impression will be.

Saving and reloading is also quite quick if you know the right hot keys, or even place the sequence of hot keys needed to load the game close each other, and this would make loading take a split second. (Default keys: L->Enter->S(:scenario)->Enter(In case you already have that save pre-selected) ).

I've seen players who save and reload to be very quick to do this key sequence, and again, all of them can be remapped. Like how I prefer to have space bar as the default confirm key.

I think my impression that this could be sped up, does not come from playing on PC but on touch device where you can't use hot keys and have to manually click on every button.

**Reply by Branikolog** [reply]
@zenseii , cannot disagree with your points. Touch devices (which a portable in common) definitely have disadvantages and are not really suitable for playing such games in my opinion. Especially with small screens.
I'm not strictly against your proposal, but we should count not only on people's impressions, but provide the best gaming experience. There should be a balance. We had a proposal of allowing to save and reload the game even during the battle. Imagine, if a player had such possibility? On paper it's a convenient improvement, which helps a player. But I believe he would stop playing the game even before finishing a single scenario being bored with thousands of reloads during the match. :)

**Reply by badtoper** [reply]
Hi,

This is my point of view as a day-one Heroes II player. Having to click multiple times to reach the save you want to load is a drag. It always has been. And if you want to load a previous save, you just do it — adding artificial steps to get there makes no sense.

The loading screen should appear directly, either with filters, set on the current game type (campaign, scenario, multiplayer), or without filters— because when you click Load during a game, 99 % of the time it’s to load a save from the current game. You would just have to clic on the main menu button if you want to load a different game type.

Also, there’s no need to display a warning that the current game will end — just like on the save screen, the Cancel button should simply cancel and return to the game.

**Reply by minuspinus** [reply]
The current behavior does not look like a penalty to the player (and there were no such goals), but as flaws in game design, such behavior cannot be found in modern games.
